src/main/java/org/ohdsi/analysis/versioning/CdmVersionUtils.java

```diff
@@ -29,13 +29,17 @@ public static void determineCdmCompatibility(CdmCompatibilitySpec compatibilityS
                 throw new IncompatibleVersionsException("Selected parameters don't have a common compatible CDM range");
             }
 
-            String overallIntersection = SemverUtils.getRangesIntersection(Arrays.asList(compatibilitySpec.getCdmVersionRange(), derivedIntersection));
-            if (overallIntersection == null) {
-                throw new IncompatibleVersionsException(
-                        String.format("User-defined CDM range (%s) doesn't match derived CDM range (%s)", compatibilitySpec.getCdmVersionRange(), derivedIntersection)
-                );
+            if (!compatibilitySpec.getCdmVersionRange().equals(VERSION_X_RANGE)) {
+                String overallIntersection = SemverUtils.getRangesIntersection(Arrays.asList(compatibilitySpec.getCdmVersionRange(), derivedIntersection));
+                if (overallIntersection == null) {
+                    throw new IncompatibleVersionsException(
+                            String.format("User-defined CDM range (%s) doesn't match derived CDM range (%s)", compatibilitySpec.getCdmVersionRange(), derivedIntersection)
```

Comment:
Should this say 'does not include derived CDM range'?  They don't need to match, the user just needs to specify a range that would intersect with the range that is derived from the expression.